test/Base_Tests/src/Data/Decimal_Spec.enso

```diff
@@ -492,6 +492,74 @@ add_specs suite_builder =
             nd2 = -d2
             nd2 . should_equal 5
 
+    suite_builder.group "remainder" group_builder->
+        group_builder.specify "should define remainder" <|
+            cases = []
+                + [[5, 3, 2], [5.0, 3.0, 2.0], [3.5, 2, 1.5], [10.5, 1.0, 0.5], [3, 1, 0], [3.0, 1.0, 0]]
+                + [[-5, 3, -2], [-5.0, 3.0, -2.0], [-3.5, 2, -1.5], [-10.5, 1.0, -0.5], [-3, 1, 0], [-3.0, 1.0, 0]]
+                + [["9223372036854775807", 10, 7], ["9223372036854775808", 10, 8], ["922337203685477580000000000008", 10, 8]]
+                + [["-9223372036854775808", 10, -8], ["-9223372036854775809", 10, -9], ["-922337203685477580000000000008", 10, -8]]
+            cases.map c->
+                base = c.at 0
+                modulus = c.at 1
+                residue = c.at 2
+                (Decimal.new base % modulus) . should_equal residue
+                (Decimal.new base % Decimal.new modulus) . should_equal residue
```

Comment:
I'd probably add a test case for something like `2.7 % 0.5`. Should it work and return `0.2` or are non-integer arguments not allowed? Either way it would be good to have a test.

Reply:
Done.